Composer/packages/client/src/pages/design/exportSkillModal/content/SelectProfile.tsx

```diff
@@ -0,0 +1,201 @@
+): string => {
+  const [version] = VERSION_REGEX.exec($schema) || [''];
+
+  let fileId = version ? `${botName}-${version.replace(/\./g, '-')}-manifest` : `${botName}-manifest`;
+  let i = -1;
+
+  while (skillManifests.some(({ id }) => id === fileId)) {
+    if (i < 0) {
+      fileId = fileId.concat(`-${++i}`);
+    } else {
+      fileId = fileId.substr(0, fileId.lastIndexOf('-')).concat(`-${++i}`);
+    }
+  }
+
+  return fileId;
+};
+
+export const SelectProfile: React.FC<ContentProps> = ({ manifest, setSkillManifest, value, onChange, projectId }) => {
+  const [publishingTargets, setPublishingTargets] = useState<PublishTarget[]>([]);
+  const [currentTarget, setCurrentTarget] = useState<PublishTarget>();
+  const { updateCurrentTarget } = useRecoilValue(dispatcherState);
+  const [endpointUrl, setEndpointUrl] = useState<string>();
+  const [appId, setAppId] = useState<string>();
+  const { id, content } = manifest;
+  const botName = useRecoilValue(botDisplayNameState(projectId));
+  const skillManifests = useRecoilValue(skillManifestsState(projectId));
+
+  const { ...rest } = content;
+  const updateCurrentProfile = useMemo(
+    () => (_e, option?: IDropdownOption) => {
+      const target = publishingTargets.find((t) => {
+        return t.name === option?.key;
+      });
+      setCurrentTarget(target);
+    },
+    [publishingTargets]
+  );
+
+  useEffect(() => {
+    if (currentTarget) {
+      setCurrentTarget(currentTarget);
+      updateCurrentTarget(projectId, currentTarget);
+      const config = JSON.parse(currentTarget.configuration);
+      setEndpointUrl(`https://${config.hostname}.azurewebsites.net/api/messages`);
+      setAppId(config.settings.MicrosoftAppId);
+
+      setSkillManifest({
+        content: {
+          ...rest,
+          endpoints: [
+            {
+              protocol: 'BotFrameworkV3',
+              name: currentTarget.name,
+              endpointUrl: `https://${config.hostname}.azurewebsites.net/api/messages`,
+              description: '<description>',
+              msAppId: config.settings.MicrosoftAppId,
+            },
+          ],
+        },
+        id: id,
+      });
+    }
+  }, [currentTarget]);
+  const isProfileValid = useMemo(
+    () => () => {
+      if (!publishingTargets) {
+        return false;
+      }
+      const filteredProfile = publishingTargets.filter((item) => {
+        const config = JSON.parse(item.configuration);
+        return (
+          config.settings.MicrosoftAppId &&
+          config.hostname &&
+          config.settings.MicrosoftAppId.length > 0 &&
+          config.hostname.length > 0
+        );
+      });
+      return filteredProfile.length > 0;
+    },
+    [publishingTargets]
+  );
+
+  const publishingOptions = useMemo(() => {
+    return publishingTargets.map((t) => ({
+      key: t.name,
+      text: t.name,
+    }));
+  }, [publishingTargets]);
+
+  const settings = useRecoilValue(settingsState(projectId));
```

Comment:
Move the useRecoilValue hook always to the top for more readability on file's dependancies

Reply:
move it to the top